packages/flame/lib/src/assets/images.dart

```diff
@@ -163,13 +183,37 @@ class Images {
   }
 }
 
-class _ImageAssetLoader {
-  _ImageAssetLoader(this.future);
+/// Individual entry in the [Images] cache.
+///
+/// This class owns the [Image] object, which can be disposed of using the
+/// [dispose] method.
+class _ImageAsset {
+  _ImageAsset.future(Future<Image> future) : _future = future;
 
-  Image? loadedImage;
-  Future<Image> future;
+  _ImageAsset.fromImage(Image image) : _image = image;
+
+  Image? get image => _image;
+  Image? _image;
+
+  Future<Image>? _future;
 
   Future<Image> retrieve() async {
-    return loadedImage ??= await future;
+    if (_future != null) {
+      _image = await _future;
```

Comment:
This doesn't have to be awaited, and `async` can be removed

Reply:
done